### PR TITLE
contrib/gomodule/redigo: don't ignore the context given as first argument (#1935)

### PR DESCRIPTION
**Pull request by lbcjbb**
### What does this PR do?

These changes stop to override the given context given as first argument to `DoContext()` when the last given argument is not a context.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Review by knusbaum — APPROVED** [review]
Good catch, regression test, and fix.

**Comment by ajgajg1134** [comment]
If you have the chance to update this branch to the latest version of `main` I would be happy to re-approve and get this merged! Alternatively if you don't have time let us know and we can try to find some time to pull this over into a separate branch and merge it that way! Thanks again for the contribution!

**Comment by lbcjbb** [comment]
It's rebased. Thanks @ajgajg1134

**Comment by ajgajg1134** [comment]
Closes #1935

**Comment by ajgajg1134** [comment]
Hello! Sorry we missed click the merge button in time before another change merged :( If possible can you please update this branch again, alternatively if you [grant us permissions to this branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) we will be able to update the branch to get this merged without any more work from your side! Thanks again!!

**Comment by lbcjbb** [comment]
Thanks

**Pull request merged**
